ci.md

```diff
@@ -0,0 +1,201 @@
+This document covers all the CI jobs that are available for Ignition and how to
+interpret their results.
+
+## Types of checks
+
+The main type of check that is performed by CI is compiling the code and running
+all automated tests. Some jobs may also run linters and other static checkers.
+Finally, we have checks that add labels, check DCO and coverage.
+
+All checks are visible at the bottom of a pull request.
+
+![PR checks](images/PR_checks.png)
+
+We use two platforms to run CI, each of them running different builds.
+
+### Jenkins server
+
+Our Jenkins server is located at https://build.osrfoundation.org/. It runs builds
+for all supported operating systems. It also runs an ABI checker for all stable
+branches.
+
+Each pull request to a stable or `main` branch will trigger:
+
+* `<library>-ci-pr_any-ubuntu-auto-amd64`: Build and test on Ubuntu Linux
+    * Builds are run inside Docker containers for Bionic or Focal.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-homebrew-amd64`: Build and test on macOS
+    * Dependencies are installed from homebrew.
+    * Library being tested is compiled using CMake and Make.
+* `<library>-ci-pr_any-windows*-amd64`: Build and test on Windows
+    * External dependencies are installed from vcpkg
+    * Library being tested and its Ignition dependencies are compiled using colcon.
+
+Pull requests to stable branches also trigger:
+
+* `<library>-abichecker-any_to_any-ubuntu_auto-amd64`: Run ABI checker
+
+### GitHub Actions
+
+GitHub actions run on GitHub's servers. Currently, it builds Linux for all
+libraries, and macOS for some of them.
+
+Every commit to any branch will trigger:
+
+* `Ubuntu CI / Ubuntu Bionic CI (push)`: Build, test and run linters on Ubuntu Bionic.
+    * Builds are run inside Docker containers.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+    * Test coverage is computed and uploaded to Codecov.
+* `Ubuntu CI / Ubuntu Focal CI (push)`: Build, test and run linters on Ubuntu Focal.
+    * Builds are run inside Docker containers.
+    * By default, dependencies are installed from stable debian binaries for
+      stable branches, and nightlies for `main` branches.
+    * Library being tested is compiled using CMake and Make.
+
+Every pull request will also trigger the exact same builds, but with the
+`pull_request` suffix. Note that pull requests from forks won't have the
+`push` checks, while pull requests from the official repositories will have
+both `pull_request` and `push`.
+
+Other checks run on GitHub Actions:
+
+* `DCO`: Checks that all commits are signed correctly for the
+    [Developer Certificate of Origin](https://developercertificate.org/).
```

Comment:
I don't know if we have information on DCO for beginner contributors, but I was very confused in the beginning, when GitHub gave me a command to run to fix DCO, and it ended up signing alllllll the commits, even ones I merged in from other people. To fix the fix, I basically rechecked out the repo... There are more helpful commands online than the one GitHub gives.

Reply:
We have some guidance in the contribution guide, so I didn't want to repeat that information here. I'll add a link: https://ignitionrobotics.org/docs/all/contributing#process